src/System.Security.Principal.Windows/tests/WindowsIdentityTests.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Security.Principal;
+using Xunit;
+
+public class WindowsIdentityTests
+{
+    [Fact]
+    public static void GetAnonymousUserTest()
+    {
+        WindowsIdentity windowsIdentity = WindowsIdentity.GetAnonymous();
+        Assert.NotNull(windowsIdentity);
+        Assert.True(windowsIdentity.IsAnonymous);
+        Assert.False(windowsIdentity.IsAuthenticated);
+    }
+
+    [Fact]
+    public static void ConstructorsAndProperties()
+    {
+        // Retrieve the Windows account token for the current user.
+        IntPtr logonToken = WindowsIdentity.GetCurrent().AccessToken.DangerousGetHandle();
```

Comment:
Could this not be problematic / prone to race conditions if the safe handle gets finalized concurrent with usage of logonToken (no AddRef/Releasing is being done to prevent that)?